marlowe-playground-client/src/MarloweEditor/State.purs

```diff
@@ -145,30 +148,63 @@ runAnalysis doAnalyze =
         lift
           $ do
               doAnalyze contract
-              lintText contents
+              processMarloweCode contents
 
-parseContract' :: String -> Maybe Contract
-parseContract' = fromTerm <=< hush <<< parseContract
+parseContract' :: String -> Maybe Extended.Contract
+parseContract' = Holes.fromTerm <=< hush <<< parseContract
 
-lintText ::
+-- This function makes all the heavy processing needed to have the Editor state in sync with current changes.
+processMarloweCode ::
   forall m.
   MonadAff m =>
   String ->
   HalogenM State Action ChildSlots Void m Unit
-lintText text = do
+processMarloweCode text = do
   analysisExecutionState <- use (_analysisState <<< _analysisExecutionState)
   let
-    parsedContract = parseContract text
+    eParsedContract = parseContract text
 
     unreachableContracts = getUnreachableContracts analysisExecutionState
 
-    (Tuple markerData additionalContext) = Linter.markers unreachableContracts parsedContract
-  markers <- query _marloweEditorPageSlot unit (Monaco.SetModelMarkers markerData identity)
-  traverse_ editorSetMarkers markers
-  -- We set the templates here so that we don't have to parse twice
-  for_ parsedContract \contractHoles ->
-    for_ ((fromTerm contractHoles) :: Maybe Contract) \contract ->
-      modifying (_analysisState <<< _templateContent) $ updateTemplateContent $ getPlaceholderIds contract
+    (Tuple markerData additionalContext) = Linter.markers unreachableContracts eParsedContract
+
+    errorMarkers =
+      markerData
+        # Array.filter (isError <<< _.severity)
+
+    -- The initial message of a hole warning is very lengthy, so we trim it before
+    -- displaying it.
+    -- see https://github.com/input-output-hk/plutus/pull/2560#discussion_r550252989
+    warningMarkers =
+      markerData
+        # Array.filter (isWarning <<< _.severity)
+        <#> \marker ->
+            let
+              trimmedMessage =
+                if String.take 6 marker.source == "Hole: " then
+                  String.takeWhile (\c -> c /= codePointFromChar '\n') marker.message
+                else
+                  marker.message
```

Comment:
Hmm... Could we add a `shortMessage` field to `IMarkerData` and modify `holeToMarker` function in `LinterText.purs` instead? I think creating the string ourselves and then dissecting it is unnecessarily contrived

Reply:
This was like this from before, I just moved that logic from one place to another (meaning, not the idea of this PR). Will keep in mind for the future, but I'm not sure if doing a `shortMessage` is possible, as this is read by Monaco directly and displayed in the editor.

Reply:
Oh, sorry, you are right, nevermind then :)